### PR TITLE
Update README for 1.0.0 release

### PR DESCRIPTION
**Pull request by propertone**
@mkomitee @twosigmajab - let me know if you have any feedback on this

### CODE REVIEW

File: README.rst
Comment (jab) [inline README.rst]: Typo: “parameter”. Also, maybe “mitigate” rather than “prevent” in the previous line, and maybe write out “denial-of-service” (or at least abbreviate DoS as I think it’s more commonly styled?)

File: docs/conf.py
Comment (jab) [inline docs/conf.py]: Might be worth changing this to use the new `__version__` attribute rather than risking it getting out of sync here. (Same goes for setup.py)

File: docs/conf.py
Comment (jab) [inline docs/conf.py]: IIUC, the copyright notice should include all years and copyright holders, not just the latest. You can abbreviate consecutive years with a hyphen like 2013-2019.

File: README.rst
Comment (jab) [inline README.rst]: Explicitly mention that the repo has moved and has a new maintainer?

**Review by jab — CHANGES_REQUESTED** [review]

File: README.rst
Comment (jab) [reply README.rst]: Also nice to include the release date corresponding to the version (and you can say “unreleased” here while this is not yet released until the release date is known).

File: docs/conf.py
Comment (jab) [reply docs/conf.py]: Also, IIUC you can add “and contributors” to cover the individuals who’ve contributed.

File: README.rst
Comment (propertone) [reply README.rst]: > Explicitly mention that the repo has moved and has a new maintainer?

I don't think that's necessary - this detail is not relevant to the module users

Comment (jab) [reply README.rst]: Certainly the change to being actively maintained again is relevant. It’s also relevant since the project has  a new canonical URL on GitHub. (Fortunately GitHub set up a redirect from the old URL.) I’m a user and at least I think this is relevant info for packages I use, and that it’s a better look to err on the side of transparency here.
UPDATE: Probably worth mentioning the license has changed too. (Note, for license changes, IIUC all copyright holders (including community contributors who haven’t signed a CLA) must agree to the license change. I hereby agree to this one, even though you already changed it:)

Comment (jab) [reply README.rst]: ~Now that you’ve added the “unreleased” to the heading, ReStructured Text is picky and requires you to extend the heading underline at least as far as the heading’s width.~ Looks good now.

File: setup.py
Comment (jab) [inline setup.py]: You can’t assume the package is importable from within its setup.py, which is run during its installation. setup.py files that single-source the version need to take a different approach. https://packaging.python.org/guides/single-sourcing-package-version/

**Review by jab — CHANGES_REQUESTED** [review]

File: README.rst
Comment (propertone) [reply README.rst]: @mlucool - thoughts on this?

File: README.rst
Comment (jab) [inline README.rst]: s/post data/request bodies

(this is done regardless of the HTTP method)

File: setup.py
Comment (jab) [inline setup.py:22]: Worth ~adding a unit test that~ updating `test_wsgi_kerberos.py` to make sure the module provides a `__version__` attribute?

File: docs/conf.py
Comment (twosigmajab) [reply docs/conf.py]: Looks like the LICENSE file needs the same treatment:
https://github.com/deshaw/wsgi-kerberos/blob/714ba41ff2b62e7d0c976eeff197839742044946/LICENSE#L1

File: setup.py
Comment (propertone) [reply setup.py:22]: Eh...the `__version__` attribute is authoritative, so I don't really think it's necessary. The package won't even install without it set.

File: README.rst
Comment (propertone) [reply README.rst]: The license change is already mentioned as part of the 1.0.0 release

Comment (jab) [reply README.rst]: Wait, based on [PyPI](http://pypi.org/project/wsgi_kerberos) and [GitHub](https://github.com/deshaw/wsgi-kerberos/releases) it looks like 1.0.0 has not been released yet. Did you intend to release 1.0.0 already? This accounts for the confusion here. Good the license change is already mentioned in any case.

Comment (jab) [reply README.rst]: Btw, would be good to include the release date (and in the meantime "unreleased") for the 1.0.0 entry as well.

File: setup.py
Comment (jab) [reply setup.py:22]: The point would be to catch that regression before it ever landed in the master branch, rather than waiting until a release to find out.

Also, with the current implementation, if `__version__` were removed, installation would fail with an unfriendly `AttributeError: 'NoneType' object has no attribute 'group'` error.

So what I was suggesting was:
- merely append `__version__` to the `from wsgi_kerberos import ...` line in `test_wsgi_kerberos.py`, so that if the attribute were ever removed, the tests would fail with an ImportError, blocking the associated PR from being merged
- ideally rewrite the `re.search(...).group(...)` in `setup.py` to give a clearer error message when it fails, or possibly even continue installing with a dummy version and just warn that version could not be detected. That way, if a bad change did get pushed straight to the master branch bypassing CI in our PR status checks (the project's GitHub settings may not forbid landing changes outside PRs), and this bug did make it into a release, it would either fail with a better error message or not even be a fatal error.

Up to you!

File: docs/conf.py
Comment (jab) [reply docs/conf.py]: Might even be worth pasting the license (with the fixed copyright notice) in a comment header in wsgi_kerberos.py and in test_wsgi_kerberos.py (several licenses recommend doing this since individual files in a project often get separated from the other files and reproduced without the LICENSE file getting included).

File: README.rst
Comment (propertone) [reply README.rst]: 1.0.0 was never released. It just corresponded to a change of license.

File: setup.py
Comment (propertone) [reply setup.py:22]: I still don't think it's necessary. I've never seen unit tests for this in other projects. The code example for exacting the version I copied from flask, but I've seen similar examples in other projects.

File: docs/conf.py
Comment (propertone) [reply docs/conf.py]: This is the standard license text for all D.E.Shaw projects. I'd leave it as is for now, but FYI @mlucool if you think we should change it.

File: README.rst
Comment (jab) [reply README.rst]: If there's no intention of releasing 1.0.0, it shouldn't appear in the list of releases in the changelog, and the changes currently listed under 1.0.0 should be merged into the not-yet-released version (currently appears as "1.1.0 (unreleased)").

Otherwise users will assume that 1.0.0 was released and get confused when they see no corresponding release for it on PyPI or GitHub.

File: setup.py
Comment (jab) [reply setup.py:22]: I guess the principle is that if it's public API, it deserves test coverage, but I get if that feels like overkill.

Just in case it's helpful to see some examples though, the popular `coverage` project has tests for its public `version_info` attribute:
https://github.com/nedbat/coveragepy/blob/34b665620889f5389178d0f37e8061c8ffa5759c/tests/test_version.py#L19 

I also test for the public `__version__` attribute (and similar package metadata) [in the unit tests for my `bidict` project](https://github.com/jab/bidict/blob/a38d51e4b559be0b82faefdfa6677788cfcd92fb/tests/test_metadata.py).

I don't see a "resolve conversation" button, but fine with me if you prefer to leave this as-is. :)

File: README.rst
Comment (propertone) [reply README.rst]: Ok fair. I'll just combine the 1.1.0 changes into the 1.0.0 release notes and actually release 1.0.0

File: docs/conf.py
Comment (propertone) [reply docs/conf.py]: Fixed the years. From what I understand the whole "and contributors" is debatable in the community. It's a fact they do license the work and not grant us copyright ownership, but the lack of saying that does negate their claims. I'd prefer to leave it as is.

Comment (mlucool) [reply docs/conf.py]: A little more guidance from google: https://opensource.google/docs/releasing/authors/

**Review by jab — APPROVED** [review]

**Comment by propertone** [comment]
Thanks for the comprehensive review! New release is published now.

**Pull request merged**
